pkg/envinfo/storage.go

```diff
@@ -149,7 +158,18 @@ func (ei *EnvInfo) DeleteStorage(name string) error {
 
 // GetStorageMountPath gets the mount path of the storage with the given storage name
 func (ei *EnvInfo) GetStorageMountPath(storageName string) (string, error) {
-	return ei.devfileObj.Data.GetVolumeMountPath(storageName, ei.componentSettings.Name)
+	containers, err := ei.GetContainers()
+	if err != nil {
+		return "", err
+	}
+
+	// since all container components have same volume mounts, we simply refer to the first container in the list
+	// refer https://github.com/openshift/odo/issues/4105 for addressing "all containers have same volume mounts"
+	paths, err := ei.devfileObj.Data.GetVolumeMountPaths(storageName, containers[0].Name)
+	if err != nil {
+		return "", err
+	}
+	return paths[0], nil
```

Comment:
@maysunfaisal @mik-dass please suggest if this should be changed.

Reply:
Looks OK to me for now. Please check the length of the containers and path before using them to avoid a panic.

Reply:
I was thinking about it just a few minutes back before I saw your comment. Ack, will change. 👍🏾

Reply:
Addressed. PTAL f8e2eb255.

Reply:
Is it a valid devfile if it doesn't have any container components?

Reply:
> Is it a valid devfile if it doesn't have any container components?

Yes, that devfile is still valid.

Reply:
In that case, I don't think we should error with like I'm doing in https://github.com/openshift/odo/pull/4465/commits/ff5a4fb0f1e6c838316a2b220c5c579105a45a2c:
```go
"invalid devfile: components.container: required value"
```
What do you suggest?

Reply:
`odo` needs a container component to work properly and thus the error is fine.